File: core/src/main/java/javabot/model/criteria/IrcUserCriteria.java
package javabot.model.criteria;

import javabot.model.IrcUser;

public class IrcUserCriteria {
  private final org.mongodb.morphia.query.Query query;
  private final String prefix;

  public IrcUserCriteria(org.mongodb.morphia.query.Query query, String prefix) {
    this.query = query;
    this.prefix = prefix + ".";
  }


  public com.antwerkz.critter.TypeSafeFieldEnd<IrcUserCriteria, IrcUser, java.lang.String> host() {
    return new com.antwerkz.critter.TypeSafeFieldEnd<IrcUserCriteria, IrcUser, java.lang.String>(this, query, prefix + "host");
  }

  public org.mongodb.morphia.query.Criteria host(java.lang.String value) {
    return new com.antwerkz.critter.TypeSafeFieldEnd<IrcUserCriteria, IrcUser, java.lang.String>(this, query, prefix + "host").equal(value);
  }

  public com.antwerkz.critter.TypeSafeFieldEnd<IrcUserCriteria, IrcUser, java.lang.String> nick() {
    return new com.antwerkz.critter.TypeSafeFieldEnd<IrcUserCriteria, IrcUser, java.lang.String>(this, query, prefix + "nick");
  }

  public org.mongodb.morphia.query.Criteria nick(java.lang.String value) {
    return new com.antwerkz.critter.TypeSafeFieldEnd<IrcUserCriteria, IrcUser, java.lang.String>(this, query, prefix + "nick").equal(value);
  }

  public com.antwerkz.critter.TypeSafeFieldEnd<IrcUserCriteria, IrcUser, java.lang.String> userName() {
    return new com.antwerkz.critter.TypeSafeFieldEnd<IrcUserCriteria, IrcUser, java.lang.String>(this, query, prefix + "userName");
  }

  public org.mongodb.morphia.query.Criteria userName(java.lang.String value) {
    return new com.antwerkz.critter.TypeSafeFieldEnd<IrcUserCriteria, IrcUser, java.lang.String>(this, query, prefix + "userName").equal(value);
  }
}

File: src/main/kotlin/javabot/model/criteria/IrcUserCriteria.kt
package javabot.model.criteria

import com.antwerkz.critter.TypeSafeFieldEnd
import javabot.model.JavabotUser
import org.mongodb.morphia.query.Criteria
import org.mongodb.morphia.query.Query

class IrcUserCriteria(private val query: Query<JavabotUser>, prefix: String) {
    private val prefix: String

    init {
        this.prefix = prefix + "."
    }

    fun host(): TypeSafeFieldEnd<IrcUserCriteria, JavabotUser, String> {
        return TypeSafeFieldEnd<IrcUserCriteria, JavabotUser, String>(
              this, query, prefix + "host")
    }

    fun host(value: String): Criteria {
        return TypeSafeFieldEnd<IrcUserCriteria, JavabotUser, String>(
              this, query, prefix + "host").equal(value)
    }

    fun nick(): TypeSafeFieldEnd<IrcUserCriteria, JavabotUser, String> {
        return TypeSafeFieldEnd<IrcUserCriteria, JavabotUser, String>(
              this, query, prefix + "nick")
    }

    fun nick(value: String): Criteria {
        return TypeSafeFieldEnd<IrcUserCriteria, JavabotUser, String>(
              this, query, prefix + "nick").equal(value)
    }

    fun userName(): TypeSafeFieldEnd<IrcUserCriteria, JavabotUser, String> {
        return TypeSafeFieldEnd<IrcUserCriteria, JavabotUser, String>(
              this, query, prefix + "userName")
    }

    fun userName(value: String): Criteria {
        return TypeSafeFieldEnd<IrcUserCriteria, JavabotUser, String>(
              this, query, prefix + "userName").equal(value)
    }
}
